### PR TITLE
Add "no_clip" to plot, text, contour and meca

### PR DESCRIPTION
**Pull request by seisman**
**Description of proposed changes**

This PR adds the *no_clip* argument (alias of **-N**) to plot, text, countour and meca.

The name *no_clip* was agreed when wrapping the *grdimage* method in https://github.com/GenericMappingTools/pygmt/pull/620#discussion_r491742142

Fixes #

**Reminders**

- [x] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [x] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

### CODE REVIEW

File: pygmt/base_plotting.py [hunk 1]
```diff
@@ -674,6 +675,18 @@ def plot(self, x=None, y=None, data=None, sizes=None, direction=None, **kwargs):
             ``'[x|y|X|Y][+a][+cl|f][+n][+wcap][+ppen]'``.
             Draw symmetrical error bars.
         {G}
+        no_clip : bool or str
+            ``'[c|r]'``
```
Comment (weiji14) [inline pygmt/base_plotting.py:679]:
```suggestion
            ``'[c|r]'``.
```

File: pygmt/base_plotting.py [hunk 2]
```diff
@@ -1178,6 +1194,8 @@ def text(
             Sets the pen used to draw a rectangle around the text string
             (see *clearance*) [Default is width = default, color = black,
             style = solid].
+        no_clip : bool
+            Do NOT clip text at map boundaries [Default will clip].
```
Comment (weiji14) [inline pygmt/base_plotting.py:1198]:
```suggestion
            Do NOT clip text at map boundaries [Default is False, i.e. will clip].
```
Not so nice having a double negative here, so some extra text to clarify default `no_clip='False'` behaviour.

File: pygmt/base_plotting.py [hunk 3]
```diff
@@ -1332,6 +1351,9 @@ def meca(
             circle is plotted at the initial location and a line connects
             the beachball to the circle. Specify pen and optionally append
             ``+ssize`` to change the line style and/or size of the circle.
+        no_clip : bool
+            Does NOT skip symbols that fall outside frame boundary specified by
+            *region* [Default plots symbols inside frame only].
```
Comment (weiji14) [inline pygmt/base_plotting.py:1356]:
```suggestion
            *region* [Default is False, i.e. plot symbols inside map frame only].
```

**Review by weiji14 — COMMENTED** [review]
Just a few minor comments, otherwise fine.

On a related note, should probably improve the documentation of `plot` at https://www.pygmt.org/v0.2.0/api/generated/pygmt.Figure.plot.html#pygmt.Figure.plot since it's quite widely used. I was going to do it in #471 but might be better to have a separate PR handling it specifically.

**Review by weiji14 — APPROVED** [review]
Ok, good to merge. The macOS test breakages seem unrelated, not sure why there's a segmentation fault on the xarray shading test?

**Comment by seisman** [comment]
> The macOS test breakages seem unrelated, not sure why there's a segmentation fault on the xarray shading test?

In PR #650, I added more tests for xarray shading. These tests work well for GMT master, but not for 6.1.1. Perhaps some of the tests have segmentation faults with GMT 6.1.1.

**Pull request merged**